### DISCUSSION TITLE
[V4 Alpha] How to prevent generating class names from ALL files in the repo? CSS file has unwanted styles

### DISCUSSION THREAD

**Discussion by saadeghi** [tailwindlabs/tailwindcss#14549] [Help]
### **Tailwind CSS 3** ([reproduction link](https://github.com/saadeghi/tw3-content-issue))
With Tailwind 3 we could control the content files with `content` config and only the class names from those files were being generated.
For example:  
[tailwind.config.js](https://github.com/saadeghi/tw3-content-issue/blob/main/tailwind.config.js)
```js
export default {
  content: [
    './public/wanted.html',
  ],
};
```
[public/wanted.html](https://github.com/saadeghi/tw3-content-issue/blob/main/public/wanted.html)
```html
<div class="bg-green-500">I only want this class name</div>
```
only generates `bg-green-500` utility:
https://github.com/saadeghi/tw3-content-issue/blob/38885a0fd7d5c48e28b671ab089817e46d9610b4/public/output.css#L557
and Ignores the other class names from other files.

### **Tailwind CSS 4 alpha** ([reproduction link](https://github.com/saadeghi/tw4-content-issue))
With Tailwind CSS 4 I want the same result, but I can't find a way. It seems like Tailwind generates utility classes for _ALL_ files and there's no way to stop it.
- I tried the new [`@source "./public/wanted.html"`](https://github.com/saadeghi/tw4-content-issue/blob/main/input.css) in CSS file.  
- I tried [`content: [ './public/wanted.html']`](https://github.com/saadeghi/tw4-content-issue/blob/main/tailwind.config.js) in JS config

But still all the utility classes from the unwanted files are included in the production CSS:  
https://github.com/saadeghi/tw4-content-issue/blob/cae799613f2fed1066a51d3e1899c44ad755ac47/public/output.css#L543

I assume currently (Tailwind CSS 4 Alpha), there's no way to tell tailwind to only generate specific styles and the CSS `@source` config and the JS `content` config are only for adding addition files on top of all the files Tailwind CSS scans automatically.  

The problem is, if there's no way to limit the files, the CSS file will be huge, specially in monorepos, all the class names form any file will be in production, even if they are not needed.

Reproduction link: https://github.com/saadeghi/tw4-content-issue which [generates 800 lines of CSS](https://github.com/saadeghi/tw4-content-issue/blob/main/public/output.css) for [a single-line HTML file with one class name](https://github.com/saadeghi/tw4-content-issue/blob/main/public/wanted.html):

**Comment by adamwathan** [comment]
Yeah no way to do it yet but it’s basically the next thing on our list 👍🏻 easy from a technical perspective, just trying to decide on the API.
